sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -112,6 +113,11 @@ class SQLContext private[sql](
    */
   def newSession(): SQLContext = new SQLContext(sharedState, isRootContext = false)
 
+  /** Pre-Load built-in User Defined Types */
+  if (isRootContext) {
```

Comment:
What if you create two root contexts?  is there a reason you don't just rely on the singlton property of `BuiltInUDT`?